src/server/jobs/new-order-jobs.ts

```diff
@@ -240,46 +250,169 @@ const newOrderCleanupQueues = createJob('new-order-cleanup-queues', '*/10 * * *
   for (const rank of ranksToClean) {
     log(`CleanupQueues :: Cleaning up ${rank} queues`);
 
-    // Fetch current image IDs from the rankType queue
-    const currentImageIds = (
-      await Promise.all(poolCounters[rank].map((pool) => pool.getAll({ limit: 10000 })))
-    )
-      .flat()
-      .map((value) => Number(value));
-
-    if (currentImageIds.length === 0) {
-      log(`CleanupQueues :: No images found for ${rank}`);
-      continue;
-    }
+    // Clean up both slots (a and b)
+    for (const slot of ['a', 'b'] as const) {
+      log(`CleanupQueues :: Cleaning up ${rank} slot ${slot}`);
```

Comment:
I'd like to suggest that we don't do this clean-up job. It's likely expensive and repeats a lot of work every 10 minutes. Instead, we should just be removing things from the queues when they become blocked or get removed. We can do this automatically using db triggers, the outbox table, and the event watching system to remove these records from redis.

Let me know if you'd like to discuss how to do this.